From 67872883a7ce2e20e0a287e9b791eaa05a830fb2 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 12 Feb 2023 17:13:08 +0200
Subject: [PATCH] [SW-120178] fwupd: differentiate pci and i2c noret

Separate the PCIe and I2C noret functions for obeying the instance
scope.
This is in preparation for implementing different noret functions for
each of the instances.

Change-Id: Idcad7943bd9426cae6cc89fd8b37f8910c1e1844
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index d5263d3..72cca40 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -81,7 +81,7 @@
 		.enable = noop,
 		.run_fast = module_i2c_fwupd_run_fast,
 		.teardown = noop,
-		.run_noreturn = module_fwupd_run_noret, /* same as FW_UPD */
+		.run_noreturn = module_i2c_fwupd_run_noret,
 	},
 	{	/* LKD module */
 		.setup = module_lkd_setup,
diff --git a/lib/common/cmd_module.h b/lib/common/cmd_module.h
index 9573925..f7ca55c 100644
--- a/lib/common/cmd_module.h
+++ b/lib/common/cmd_module.h
@@ -96,6 +96,7 @@
 enum cmd_check_res module_i2c_fwupd_check(void);
 void module_i2c_fwupd_run_fast(void);
 void module_i2c_fwupd_run(void);
+void module_i2c_fwupd_run_noret(void);
 bool i2c_fwupd_owned(u32 offset);
 u32 i2c_fwupd_readl(u32 offset);
 void i2c_fwupd_writel(u32 offset, u32 val);
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 65762ad..44e4471 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -763,8 +763,18 @@
 	__module_fwupd_run(&i2c_state);
 }
 
-void module_fwupd_run_noret(void)
+void __module_fwupd_run_noret(struct fwupd_state *state)
 {
 	hl_ts_shutdown();
 	fit_run_image();
 }
+
+void module_fwupd_run_noret(void)
+{
+	__module_fwupd_run_noret(&state);
+}
+
+void module_i2c_fwupd_run_noret(void)
+{
+	__module_fwupd_run_noret(&i2c_state);
+}
